util/grub.d/00_menu_auto_hide.in

```diff
@@ -1,15 +1,18 @@
 #! /bin/sh
 
-# Disable / skip generating menu-auto-hide config parts on serial terminals
-for x in ${GRUB_TERMINAL_INPUT} ${GRUB_TERMINAL_OUTPUT}; do
-  case "$x" in
-    serial*)
-      exit 0
-      ;;
-  esac
-done
-
 cat << EOF
+# Boot Counting
+if [ "\${boot_success}" = "0"]; then
+  if  [ "\${boot_counter}" = "0" ]; then
+    set default=1
+    set fallback_active=true
+    save_env fallback_active
```

Comment:
Probably you don't need this `fallback_active` variable, I would just use the `boot_counter` variable as an indication that the fallback entry was booted. For example you can set here `boot_counter` to -1 or unset it so user-space knows that the default entry failed to boot.

Reply:
ack